line/backend/src/functions/webhook.ts

```diff
@@ -1,11 +1,56 @@
 import { messagingApi } from '@line/bot-sdk';
 import type { Request, Response } from 'express';
-import { config } from '../config';
+import { config } from '../config.js';
+import * as https from "https";
 
 const { MessagingApiClient } = messagingApi;
 
 const client = new MessagingApiClient(config.line.messagingApiClient);
 
+type User = {
+    // user_id: string; //TODO: 【要検討】talk履歴からだと相手のuser_idは取得できない
+    name: string; // ユーザー名
+};
+
+type TalkHistory = {
+date: string;
+user: User;
+message: string;
+};
+
+function parseTalkHistory(talk: string): TalkHistory[] {
+const lines = talk.split("\n");
+const talkHistory: TalkHistory[] = [];
+let talkDate: string | null = null;
+
+lines.forEach(line => {
+    const trimmedLine = line.trim();
+    // 日付
+    const dateMatch = trimmedLine.match(/^(\d{4}\/\d{2}\/\d{2})/);
+    if (dateMatch) {
+    talkDate = dateMatch[1].replace(/\//g, "-"); // YYYY-MM-DD
+    return;
+    }
+
+    // メッセージ（例: "22:07   Test    おはよう"）
+    const messageMatch = trimmedLine.match(/^(\d{2}:\d{2})\s+([^ ]+)?\s+(.+)$/);
```

Comment:
`[^ ]` これだと、半角スペースだけが対象なので下記のように `\S` を使って、空白以外を指定した方がよい気がしてます
ただ、トーク履歴の区切り文字によります。
確認したところ、タブのようなので `\t`でタブだけを対象にしたほうがよさそうです
現状の実装だと、メッセージの前に空白前を詰めていたらバグりそうです


```suggestion
    const messageMatch = trimmedLine.match(/^(\d{2}:\d{2})\t+([^\t]+)?\t+(.+)$/);
```